tests/daemon/test-circular-file.c

```diff
@@ -609,11 +611,158 @@ test_circular_file_shrink (Fixture      *fixture,
   g_object_unref (circular_file_2);
 }
 
+static void
+test_circular_file_broken_metadata_file (Fixture      *fixture,
+                                         const gchar  *metadata_file_contents,
+                                         gssize        metadata_file_length,
+                                         GKeyFileError expected_error_code)
+{
+  const gsize max_size = 1024;
+  g_autofree gchar *data_file_contents = g_malloc (max_size);
+  g_autofree gchar *metadata_file_path =
+    g_strconcat (fixture->data_file_path, METADATA_EXTENSION, NULL);
+  g_autoptr(GError) error = NULL;
+  gboolean ret;
+
+  /* Fill the data file with junk. */
+  memset (data_file_contents, '*', max_size);
+  ret = g_file_set_contents (fixture->data_file_path, data_file_contents, max_size, &error);
+  g_assert_no_error (error);
+  g_assert_true (ret);
+
+  /* Initialize the metadata file to the desired incorrect contents */
+  ret = g_file_set_contents (metadata_file_path, metadata_file_contents, metadata_file_length, &error);
+  g_assert_no_error (error);
+  g_assert_true (ret);
+
+  g_autoptr(EmerCircularFile) circular_file = NULL;
+
+  circular_file = emer_circular_file_new (fixture->data_file_path, max_size,
+                                          /* reinitialize */ FALSE,
+                                          &error);
+  g_assert_null (circular_file);
```

Comment:
If you silently deal with missing groups/keys you'll receive a valid circular file here, so these assertions won't longer be valid, I think. Might need to refactor this code a bit in that case

Reply:
Good point – I'll split this test helper in two.